subsys/bluetooth/services/fast_pair/fp_gatt_service.c

```diff
@@ -57,11 +68,32 @@ enum fp_msg_type {
 	FP_MSG_ACTION_REQ               = 0x10,
 };
 
-struct msg_key_based_pairing_req {
+enum additional_action {
+	ADDITIONAL_ACTION_NOTIFY_PN_BIT_POS,
+	ADDITIONAL_ACTION_COUNT,
+};
+
+struct msg_kbp_req_data {
+	uint8_t seeker_address[BT_ADDR_SIZE];
+};
+
+struct msg_action_req_data {
+	uint8_t msg_group;
+	uint8_t msg_code;
+	uint8_t additional_data_len_or_id;
```

Comment:
Maybe more readable would be to use union

Reply:
We've discussed it with @MarekPieta. It generates some problems at parsing stage, because we need to check flags to determine which field it is and therefore we need to make `enum fp_action_flags` public. We've decided that it's not wort it.